pkgs/applications/misc/redshift/default.nix

```diff
@@ -9,18 +9,25 @@
 , withGeolocation ? true
 , withCoreLocation ? withGeolocation && stdenv.isDarwin, CoreLocation, Foundation, Cocoa
 , withGeoclue ? withGeolocation && stdenv.isLinux, geoclue
+
+, wlrootsSupport ? false
 }:
 
 stdenv.mkDerivation rec {
-  name = "redshift-${version}";
+  name = "redshift${stdenv.lib.optionalString wlrootsSupport "-wlroots"}-${version}";
   version = "1.12";
 
-  src = fetchFromGitHub {
+  src = fetchFromGitHub (if wlrootsSupport then {
```

Comment:
It's just a single patch. If it applies to the lastest release of the proper redshift, I'd rather do that with `patches = [ fetchpatch { url = "https://github.com/minus7/redshift/commit/eecbfedac48f827e96ad5e151de8f41f6cd3af66.diff"; sha256 = ""; } ];` rather than substituting the entire soruce tree.

Reply:
it applies to the latest commit on master, which is 24 commits ahead of redshift v1.2

Reply:
i agree that it's more elegant, but won't this be hard to maintain anyway in case redshift-wlroots falls behind and conflicts develop?

Reply:
@JohnAZoidberg here you go, anyway